Assets/CustomAssets/Textures/MovingWater.shader

```diff
@@ -20,97 +22,116 @@
 			#include "UnityCG.cginc"
 			#include "UnityLightingCommon.cginc" // for _LightColor0
 
-			fixed4 _CellColor;
+			fixed4 _MainColor;
 			fixed4 _SecondaryColor;
 			float _Speed;
 			float _CellAmount;
 			float _Shininess;
-			
+			int _Ripples;
+			int _Offset;
+
+			struct appdata
+			{
+				float4 vertex : POSITION;
+				float3 normal : NORMAL;
+				float2 uv : TEXCOORD0;
+			};
+
 			struct v2f
 			{
 				float2 uv : TEXCOORD0;
 				float4 vertex : SV_POSITION;
-				fixed4 col : COLOR0; // diffuse lighting color
+				float3 worldPos : TEXCOORD1; //world position,
+				float3 worldNormal : TEXCOORD2; //normal vector in world space
 				float3 normal : NORMAL;
 			};
 
-			v2f vert(appdata_base v)
+			v2f vert(appdata v)
 			{
 				v2f o;
-				float _MaxDisplacement = 1;
-
 
 				o.vertex = v.vertex;
-				o.vertex.y += sin(0.5 * (v.vertex.x + _Time.w)) / 10;
-				o.vertex.y += cos(0.5 * (v.vertex.z + _Time.w)) / 9;
 
-				o.vertex = UnityObjectToClipPos(o.vertex);
+				//position of vertices, transforming from object to camera space
+				o.vertex = UnityObjectToClipPos(v.vertex);
 
-				o.uv = v.texcoord;
+				o.uv = v.uv;
 
+				//get vertex normal in world space
+				o.worldPos = mul(unity_ObjectToWorld, v.vertex);
+				o.worldNormal = normalize(mul(v.normal, (float3x3)unity_WorldToObject));
 
+				//making the waves - only adjusting the Y axis (height)
+				o.vertex.y += sin(0.5 * (v.vertex.x + _Time.w)) / 10;
+				o.vertex.y += cos(0.5 * (v.vertex.z + _Time.w)) / 9;
 
-				// get vertex normal in world space
-				half3 worldNormal = UnityObjectToWorldNormal(v.normal);
-
-				float4x4 modelMatrix = unity_ObjectToWorld;
-				float3 viewport = normalize(_WorldSpaceCameraPos - mul(modelMatrix, v.vertex).xyz);
-				float3 lightDirection = normalize(_WorldSpaceLightPos0.xyz);
-				float3 reflectionVector = reflect(-lightDirection, worldNormal);
-
-				// standard diffuse (Lambert) lighting
-				fixed4 nl = max(0, dot(worldNormal, lightDirection));
-
-				// factor in the light color
-				fixed4 diffuse = nl * _LightColor0;
-
-				// find specular dot product
-				fixed4 specular = pow(max(0.0, dot(reflectionVector, viewport)), _Shininess) * _LightColor0;
-				o.col = specular + diffuse ;
 				return o;
 			}
 
-			float2 random2(float2 p)
+			float2 random2d(float2 rand) //takes a vector2 to give an output of random vector2 between 0 and 1
 			{
-				//return frac(sin(float2(dot(p,float2(117.12,341.7)),dot(p,float2(269.5,123.3))))*43458.5453);
-				return frac(sin(float2(dot(p, float2(127.1, 311.7)), dot(p, float2(269.5, 183.3))))*43758.5453);
+				return frac(sin(float2(dot(rand, float2(127.1, 311.7)), dot(rand, float2(269.5, 183.3))))*43758.5453);
 			}
 
 			fixed4 frag(v2f i) : SV_Target
 			{
+
 				fixed4 col = _SecondaryColor;
```

Comment:
same thing as above mentioned ☝️

Reply:
I'm not sure what you mean here

Reply:
I mean it's same "issue", which I mentioned in the comment above

Reply:
*I'm not sure which part of the code you mean here as a single line is commented

Reply:
I mean exactly that one line 😄 `fixed4 col = _SecondaryColor;` in `MovingWater.shader` line 79